pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -361,13 +352,17 @@ public Representation post(Representation entity) {
           // list of FileItems
           items = upload.parseRequest(getRequest());
 
-          for (final Iterator<FileItem> it = items.iterator(); it.hasNext() && !found;) {
-            final FileItem fi = it.next();
-            if (fi.getFieldName() != null) {
-              found = true;
-              dataFile = new File(tempDir, fi.getFieldName());
-              fi.write(dataFile);
+          for (FileItem fileItem : items) {
```

Comment:
Can multiple pushes of the same file at the same time cause any concurrency issue here?